src/com/carolinarollergirls/scoreboard/defaults/DefaultScoreBoardModel.java

```diff
@@ -185,144 +179,234 @@ public void startJam() {
+		ClockModel pc = getClockModel(Clock.ID_PERIOD);
+		ClockModel jc = getClockModel(Clock.ID_JAM);
+		ClockModel tc = getClockModel(Clock.ID_TIMEOUT);
+
+		if (pc.isTimeAtEnd() && !pc.isRunning() && !jc.isRunning() && !tc.isRunning()) {
 			requestBatchStart();
-			if (pc.isTimeAtEnd()) {
-				ic.resetTime(); //TODO: add a rule to make this optional
-				ic.start();
-			} else {
-				lc.resetTime();
-				lc.start();
-			}
-			tc.stop();
+			setInPeriod(false);
+			setOfficialScore(false);
+			_endLineup();
+			_startIntermission();
 			requestBatchEnd();
 		}
 	}
-	private void _startLineup() {
-		synchronized (runLock) {
-			createSnapshot(ACTION_LINEUP);
-			
-			ClockModel lc = getClockModel(Clock.ID_LINEUP);
-			ClockModel ic = getClockModel(Clock.ID_INTERMISSION);
+	private void _startJam() {
+		ClockModel pc = getClockModel(Clock.ID_PERIOD);
+		ClockModel jc = getClockModel(Clock.ID_JAM);
+
+		requestBatchStart();
+		_endIntermission(false);
+		_endTimeout();
+		_endLineup();
+		setInPeriod(true);
+		pc.start();
+		jc.startNext();
+
+		getTeamModel("1").startJam();
+		getTeamModel("2").startJam();
+		requestBatchEnd();
+	}
+	private void _endJam(boolean force) {
+		ClockModel pc = getClockModel(Clock.ID_PERIOD);
+		ClockModel jc = getClockModel(Clock.ID_JAM);
 
-			ic.stop();
+		if (!jc.isRunning() && !force) { return; }
+		
+		requestBatchStart();
+		jc.stop();
+		getTeamModel("1").stopJam();
+		getTeamModel("2").stopJam();
+		setInOvertime(false);
 
-			requestBatchStart();
-			lc.resetTime();
-			lc.start();
-			requestBatchEnd();
+		if (pc.getTimeRemaining() < 30000) {
+			restartPcAfterTimeout = true;
+		}
+		if (pc.isRunning()) {
+			_startLineup();
+		} else {
+			_possiblyEndPeriod();
 		}
+		requestBatchEnd();
 	}
+	private void _startLineup() {
+		ClockModel lc = getClockModel(Clock.ID_LINEUP);
 
-	public void timeout() { timeout(null); }
-	public void timeout(TeamModel team) { timeout(team, false); }
-	public void timeout(TeamModel team, boolean review) {
-		synchronized (runLock) {
-			requestBatchStart();
-			createSnapshot(ACTION_TIMEOUT);
-			
-			ClockModel tc = getClockModel(Clock.ID_TIMEOUT);
-			ClockModel pc = getClockModel(Clock.ID_PERIOD);
-			ClockModel jc = getClockModel(Clock.ID_JAM);
-			ClockModel lc = getClockModel(Clock.ID_LINEUP);
+		requestBatchStart();
+		_endIntermission(false);
+		setInPeriod(true);
+		lc.startNext();
+		requestBatchEnd();
+	}
+	private void _endLineup() {
+		ClockModel lc = getClockModel(Clock.ID_LINEUP);
 
-			if (null==team && tc.isRunning() && getTimeoutOwner() == "") {
+		requestBatchStart();
+		lc.stop();
+		requestBatchEnd();
+	}
+	private void _startTimeout() {
+		ClockModel pc = getClockModel(Clock.ID_PERIOD);
+		ClockModel tc = getClockModel(Clock.ID_TIMEOUT);
+
+		requestBatchStart();
+		if (tc.isRunning()) {
+			//TODO: Make Official Timeout its own button that calls setTimeoutType()
+			if (getTimeoutOwner()=="") {
 				setTimeoutOwner("O");
```

Comment:
It's descriebd in the Wiki since 3.9.3.

Reply:
Can you point me to where the time out owner can have the value "O" on the wiki? This is an implementation detail I'd expect to be mentioned in a comment, as otherwise I'd expect it to be "1" or "2".

Reply:
Oh, you mean the fact that "O" is OTO. That is indeed not on the Wiki AFAIK.

Reply:
It's not in the WS (that comes out as bools), so it probably shouldn't be on the wiki as it's internals only of interest to a few of us. A quick comment somewhere, or appropriately named consts, would do the trick I think.

Reply:
It's in the WS in ScoreBoard.TimeoutOwner and is used in the WS display.

Reply:
Ah, I missed that. That's a todo for me then.